terraform/variables.tf

```diff
@@ -0,0 +1,46 @@
+variable "app_name" {
+  description = "Application name"
+  type        = string
+  default     = "opensearch"
+}
+
+variable "channel" {
+  description = "Charm channel"
+  type        = string
+  default     = null
+}
+
+variable "base" {
+  description = "Charm base (old name: series)"
+  type        = string
+  default     = "ubuntu@22.04"
+}
+
+variable "config" {
+  description = "Map of charm configuration options"
+  type        = map(string)
+  default     = {}
+}
+
+variable "model_name" {
+  description = "Model name"
+  type        = string
+}
```

Comment:
The spec was changed to use the name `model` for the input variable to better communicate the fact that it should a tied to a resource, and not just any string.

Also, I have heard that there will be changes to the provider moving towards UUIDs, so using `model` here would also better fit with that change in the future.